synapse/storage/databases/main/roommember.py

```diff
@@ -350,6 +350,39 @@ def _get_rooms_for_local_user_where_membership_is_txn(
 
         return results
 
+    @cached()
```

Comment:
Do we need to invalidate this cache ever?

Reply:
Yep! And after looking at the places this cache needs to be invalidated... it's a few.

It was a nice thought, but since this is quite a quick query I'm minded to not block a lot of work on a cache here :l